pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -210,266 +210,275 @@ let
-    for x in ${docdir "$doc"}"/html/src/"*.html; do
-      remove-references-to -t $out $x
-    done
-    mkdir -p $doc
-    ''}
-    ${optionalString enableSeparateDataOutput "mkdir -p $data"}
-
-    runHook postInstall
-  '';
-
-  passthru = passthru // {
-
-    inherit pname version;
-
-    compiler = ghc;
-
-
-    getBuildInputs = {
-      inherit propagatedBuildInputs otherBuildInputs allPkgconfigDepends;
-      haskellBuildInputs = isHaskellPartition.right;
-      systemBuildInputs = isHaskellPartition.wrong;
+  drv // {
+    env = buildHaskellPackages.shellFor {
+      packages = p: [ drv ];
     };
-
-    isHaskellLibrary = isLibrary;
-
-    # TODO: ask why the split outputs are configurable at all?
-    # TODO: include tests for split if possible
-    # Given the haskell package, returns
-    # the directory containing the haddock documentation.
-    # `null' if no haddock documentation was built.
-    # TODO: fetch the self from the fixpoint instead
-    haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
-
-  };
-
-  meta = { inherit homepage license platforms; }
-         // optionalAttrs broken               { inherit broken; }
-         // optionalAttrs (description != "")  { inherit description; }
-         // optionalAttrs (maintainers != [])  { inherit maintainers; }
-         // optionalAttrs (hydraPlatforms != null) { inherit hydraPlatforms; }
-         ;
-
-}
-// optionalAttrs (preCompileBuildDriver != "")  { inherit preCompileBuildDriver; }
-// optionalAttrs (postCompileBuildDriver != "") { inherit postCompileBuildDriver; }
-// optionalAttrs (preUnpack != "")      { inherit preUnpack; }
-// optionalAttrs (postUnpack != "")     { inherit postUnpack; }
-// optionalAttrs (patches != [])        { inherit patches; }
-// optionalAttrs (patchPhase != "")     { inherit patchPhase; }
-// optionalAttrs (preConfigure != "")   { inherit preConfigure; }
-// optionalAttrs (postConfigure != "")  { inherit postConfigure; }
-// optionalAttrs (preBuild != "")       { inherit preBuild; }
-// optionalAttrs (postBuild != "")      { inherit postBuild; }
-// optionalAttrs (doBenchmark)          { inherit doBenchmark; }
-// optionalAttrs (checkPhase != "")     { inherit checkPhase; }
-// optionalAttrs (preCheck != "")       { inherit preCheck; }
-// optionalAttrs (postCheck != "")      { inherit postCheck; }
-// optionalAttrs (preInstall != "")     { inherit preInstall; }
-// optionalAttrs (installPhase != "")   { inherit installPhase; }
-// optionalAttrs (postInstall != "")    { inherit postInstall; }
-// optionalAttrs (preFixup != "")       { inherit preFixup; }
-// optionalAttrs (postFixup != "")      { inherit postFixup; }
-// optionalAttrs (dontStrip)            { inherit dontStrip; }
-// optionalAttrs (hardeningDisable != []) { inherit hardeningDisable; }
-// optionalAttrs (stdenv.buildPlatform.libc == "glibc"){ LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive"; }
-)
+  }
```

Comment:
Recommended to hide whitespace changes (Click on Diff settings above).

I don't see any other way than this to add `.env` to passthru. Because it itself depends on the thing.